### PR TITLE
Streaming update is handled in frontend instead of ChatHub

### PR DESCRIPTION
**Pull request by Jimmy-Z**
I think frontend should have more control over how it's updated.

### CODE REVIEW

**Review by acheong08 — COMMENTED** [review]
There are some useful information as part of the non-streamed response that people might want to use such as suggestions

File: src/EdgeGPT.py [hunk 1]
```diff
@@ -201,21 +176,18 @@ async def ask_stream(self, prompt: str) -> str:
...
                 if obj is None or obj == "":
                     continue
                 response = json.loads(obj)
                 if response.get("type") == 1:
-                    cache_len = len(stream_cache)
-                    adaptive_card = response["arguments"][0]["messages"][0]["adaptiveCards"][0]["body"][0]["text"]
-                    stream_cache = adaptive_card
-                    adaptive_card = adaptive_card[cache_len:]
-                    yield adaptive_card
+                    yield False, response["arguments"][0]["messages"][0]["adaptiveCards"][0]["body"][0]["text"]
```
Comment (acheong08) [inline src/EdgeGPT.py:187]: I want it to stream token by token

Comment (acheong08) [reply src/EdgeGPT.py:187]: It makes it easier than having to clear out old outputs

Comment (Jimmy-Z) [reply src/EdgeGPT.py:187]: That update logic is moved to main(), not removed.

Comment (acheong08) [reply src/EdgeGPT.py:187]: Makes sense. Ok got it

**Review by acheong08 — APPROVED** [review]
Looks about right. I'll try to run this and see if there are any flaws

File: src/EdgeGPT.py (hunk 1 above)
Comment (Jimmy-Z) [reply src/EdgeGPT.py:187]: I'm guessing microsoft/openai designed it like that for a reason, in the future the update might change text from the previous update.

**Comment by acheong08** [comment]
Perfect. It works

**Pull request merged**